charon/src/translate_functions_to_im.rs

```diff
@@ -545,374 +545,341 @@ fn translate_projection<'tcx>(
     return (projection, path_type);
 }
 
-/// Translate some cases of a constant operand.
-///
-/// A constant operand is usually an instance of a primitive type (bool, char,
-/// integer...). However, it may also be an instance of a degenerate ADT or
-/// tuple (if an ADT has only one variant and no fields, it is a constant,
-/// and unit is encoded by MIR as a 0-tuple). This function handles the
-/// primitive type cases: the other cases should have been filtered and
-/// handled elsewhere.
```

Comment:
I think the comment about the degenerate ADTs or tuples is useful and should be preserved (in the documentation for `translate_evaluated_operand_constant` maybe?)

Reply:
Ah yes, I put it on the `match` of `translate_constant_scalar_value` (where all those cases are handled).

Reply:
Ok thanks, hadn't noticed!